File: Language.Implementation.Patterns/python/ch02_basic_parsing_patterns/p02_lexer/Token.py

class Token:
    def __init__(self, type, text):
        self.type = type
        self.text = text

    def __str__(self):
        from ListLexer import ListLexer
        tname = ListLexer.tokenNames[self.type]
        return "<'" + self.text + "'," + tname + ">"

File: Language.Implementation.Patterns/cxx/parsing/recursive-descent/Token.cpp
#include "Token.hpp"

#include <sstream>
#include <vector>

#include "ListLexer.hpp"

std::string Token::toString() const {
    std::string tname = ListLexer::tokenNames[type];
    std::ostringstream oss;
    oss << "<'" << text << "'," << tname << ">";
    return oss.str();
}
